File: app/src/main/java/eu/kanade/tachiyomi/data/mangasync/MangaSyncManager.java
package eu.kanade.tachiyomi.data.mangasync;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

import eu.kanade.tachiyomi.data.mangasync.base.MangaSyncService;
import eu.kanade.tachiyomi.data.mangasync.services.MyAnimeList;

public class MangaSyncManager {

    private List<MangaSyncService> services;
    private MyAnimeList myAnimeList;

    public static final int MYANIMELIST = 1;

    public MangaSyncManager(Context context) {
        services = new ArrayList<>();
        myAnimeList = new MyAnimeList(context);
        services.add(myAnimeList);
    }

    public MyAnimeList getMyAnimeList() {
        return myAnimeList;
    }

    public List<MangaSyncService> getSyncServices() {
        return services;
    }

    public MangaSyncService getSyncService(int id) {
        switch (id) {
            case MYANIMELIST:
                return myAnimeList;
        }
        return null;
    }

}

File: app/src/main/java/eu/kanade/tachiyomi/data/mangasync/MangaSyncManager.kt
package eu.kanade.tachiyomi.data.mangasync

import android.content.Context
import eu.kanade.tachiyomi.data.mangasync.anilist.Anilist
import eu.kanade.tachiyomi.data.mangasync.myanimelist.MyAnimeList

class MangaSyncManager(private val context: Context) {

    companion object {
        const val MYANIMELIST = 1
        const val ANILIST = 2
    }

    val myAnimeList = MyAnimeList(context, MYANIMELIST)

    val aniList = Anilist(context, ANILIST)

    // TODO enable anilist
    val services = listOf(myAnimeList)

    fun getService(id: Int) = services.find { it.id == id }

}
